modin/core/execution/ray/common/utils.py

```diff
@@ -217,7 +217,17 @@ def initialize_ray(
         _move_stdlib_ahead_of_site_packages
     )
     ray.worker.global_worker.run_function_on_all_workers(_import_pandas)
-    num_cpus = int(ray.cluster_resources()["CPU"])
+    num_cpus = ray.cluster_resources().get("CPU", None)
+    if num_cpus is None:
+        warnings.warn(
+            "The current Ray cluster does not have any CPU Resources.\nModin uses the number of "
+            + "CPUs to determine how many partitions to create.\nNumber of partitions defaulting to"
+            + " 4. To update, run the following python code:\n\tfrom modin.config import "
+            + "NPartitions\n\tNPartitions.put(desired_num_cpus)"
+        )
+        num_cpus = 4
```

Comment:
I think we should add an inline comment why we set `num_cpus=4` by default. I remember there was a question why we did that and whether we should have changed it. Also, it would be good to reflect this in the docs somewhere.

Reply:
Btw, is such kind of a scenario applicable to Dask?

Reply:
+1 we need documentation.

I think this might also be applicable to Dask, but we'd need to test it.

Reply:
I'm happy to add a comment here! Before that: Is 4 a good heuristic? I picked it bc most personal laptops have 4 cores/it seemed like a good number, but I'd prefer to have a better reason behind how we treat this case, so I'd appreciate y'all's insight into how we can better handle this case!